packages/flutter_tools/lib/src/base/version.dart

```diff
@@ -119,3 +119,35 @@ class Version implements Comparable<Version> {
   @override
   String toString() => _text;
 }
+
+/// Returns true if [targetVersion] is within the range [min] and [max]
+/// inclusive by default.
+///
+/// [min] and [max] are evaluated by [Version.parse(text)].
+///
+/// Pass [inclusiveMin] = false for greater than and not equal to min.
+/// Pass [inclusiveMax] = false for less than and not equal to max.
+bool isWithinVersionRange(
```

Comment:
@christopherfujino Are the constraints on our use of `pub_semver` in the tool? I know we have a dependency on it already, and using its `Version` and `VersionRange` would eliminate the need for this code. (I'm not sure if there's a reason for the existence of our `Version` rather than using that one, other than historical.)

Reply:
Not that I know of, using pub_semver sounds good to me.

Reply:
Actually, I'm not sure that there aren't some pub-specific hacks in that package. @reidbaker can you add a TODO to investigate using pub_semver?

Reply:
Done